crates/matrix-sdk-base/src/event_cache_store/integration_tests.rs

```diff
@@ -0,0 +1,193 @@
+            "media still there after removing"
+        );
+
+        // Let's add the media again.
+        self.add_media_content(&request_file, content.clone())
+            .await
+            .expect("adding media again failed");
+
+        assert_eq!(
+            self.get_media_content(&request_file).await.unwrap().as_ref(),
+            Some(&content),
+            "media not found after adding again"
+        );
+
+        // Let's add the thumbnail media.
+        self.add_media_content(&request_thumbnail, thumbnail_content.clone())
+            .await
+            .expect("adding thumbnail failed");
+
+        // Media's thumbnail is present.
+        assert_eq!(
+            self.get_media_content(&request_thumbnail).await.unwrap().as_ref(),
+            Some(&thumbnail_content),
+            "thumbnail not found"
+        );
+
+        // Let's add another media with a different URI.
+        self.add_media_content(&request_other_file, other_content.clone())
+            .await
+            .expect("adding other media failed");
+
+        // Other file is present.
+        assert_eq!(
+            self.get_media_content(&request_other_file).await.unwrap().as_ref(),
+            Some(&other_content),
+            "other file not found"
+        );
+
+        // Let's remove media based on URI.
+        self.remove_media_content_for_uri(uri).await.expect("removing all media for uri failed");
+
+        assert!(
+            self.get_media_content(&request_file).await.unwrap().is_none(),
+            "media wasn't removed"
+        );
+        assert!(
+            self.get_media_content(&request_thumbnail).await.unwrap().is_none(),
+            "thumbnail wasn't removed"
+        );
+        assert!(
+            self.get_media_content(&request_other_file).await.unwrap().is_some(),
+            "other media was removed"
+        );
+    }
+}
+
+/// Macro building to allow your `EventCacheStore` implementation to run the
+/// entire tests suite locally.
+///
+/// You need to provide a `async fn get_event_cache_store() ->
+/// EventCacheStoreResult<impl EventCacheStore>` providing a fresh event cache
+/// store on the same level you invoke the macro.
+///
+/// ## Usage Example:
+/// ```no_run
+/// # use matrix_sdk_base::event_cache_store::{
+/// #    EventCacheStore,
+/// #    MemoryStore as MyStore,
+/// #    Result as EventCacheStoreResult,
+/// # };
+///
+/// #[cfg(test)]
+/// mod tests {
+///     use super::{EventCacheStore, EventCacheStoreResult, MyStore};
+///
+///     async fn get_event_cache_store(
+///     ) -> EventCacheStoreResult<impl EventCacheStore> {
+///         Ok(MyStore::new())
+///     }
+///
+///     event_cache_store_integration_tests!();
+/// }
+/// ```
+#[allow(unused_macros, unused_extern_crates)]
+#[macro_export]
+macro_rules! event_cache_store_integration_tests {
+    () => {
+        mod event_cache_store_integration_tests {
+            $crate::event_cache_store_integration_tests!(@inner);
```

Comment:
Does having an `@inner` macro buy us anything, or could we inline the below macro-expanded block here instead? (and have a single rule for this macro)

Reply:
I don't think so, I'll remove it.

Reply:
Removed